[FEATURE] Emergency Response Team

Steelpoint opened the pull request:
This PR see's the addition of Nanotrasen's asset recovery squad, the Emergency Response Team.

I don't expect this PR to be merged as of now due to the multitude of other PR's that would conflict with this PR before this PR is good to merge, however I do intend to use this to garner feedback.

Do note that this squad is a admin event only, however there is a "ERT Spawn Button" available to admins to easily spawn in a squad.

Features

---

Adds a ERT ready room to the CentCom Z-Level. This room is where the ERT will ready their gear and then move to the station onboard the CentCom shuttle.

![screenshot 2015-01-01 15 19 57](https://dl.dropboxusercontent.com/s/jnvsdg10l0rf8uw/Screenshot%202015-01-01%2018.25.45.png?dl=0)

You would notice that there are two back rooms. They can only be opened by a admin. They are:

Emergency Response Team Equipment Room (Upper Room)
This holds additional heavier equipment meant for any ERT that is going into a uncertain situation that may require more firepower. 

Death Squad Equipment Room
Meant as a alternative to the way the Death Squad is deployed, contains all the needed DS equipment employed, as well as two Marauder mechs.

The ERT consists of three specialised roles and a commander, all equipment is subject to change and revision, they are:

![](https://cloud.githubusercontent.com/assets/6595389/5591991/7b22770a-91e1-11e4-8693-b4736027be3e.png)
- ERT Commander: The squad leader, has access to some basic items from all three fields. The Commanders role is to primarily command the squad during the mission.
  - Autorevolver
  - 2x .357 speedloaders
  - Stun Baton
  - Flash
  - Night Vision SecHUD
  - First Aid Kit
  - Box of Handcuffs
  - Intelicard

![secert](https://cloud.githubusercontent.com/assets/6595389/5588781/5d11dd00-914b-11e4-8862-9e54be2cf94d.png)
- Security Officer: Employs more robust non-lethal firepower, defends the ERT from possible threats on the station. Equipped with:
  - Pulse Carbine (20 Pulse/Kill/Stun shots)
  - Stun Baton
  - Flash
  - Telescopic Shield
  - Security Belt
  - SecHUDs
  - Box of Flashbangs
  - Box of Handcuffs

![engert](https://cloud.githubusercontent.com/assets/6595389/5588783/62eaa87e-914b-11e4-8c54-e3ec5fe44884.png)
- Engineering Officer: Has a large array of repairing tools designed to help quickly repair critical damage. Equipped with:
  - Military RCD: 200 cartridge RCD
  - 3x Matter Cartridge
  - Stun Baton
  - 2x Metal Foam Grenades
  - Glass and Metal Sheet
  - Toolbelt
  - Wires
  - Optical Mesons

![medert](https://cloud.githubusercontent.com/assets/6595389/5588785/66906e14-914b-11e4-8420-068a1d31c104.png)
- Medical Officer: Equipped with a robust selection of healing equipment, ready to preform urgent triage on wounded and dying personal. Equipped with:
  - Adminordrazine Combat Injector: 190 units adminordrazine, rapid healing.
  - Stun Baton
  - All four basic first aid kits
  - Body Bag Box
  - Medical Belt
  - Medical HUD

In addition, all ERT Officers are equipped with the following as standard issue.
- Centcom Officer clothing
- Combat Boots
- Combat Gloves
- Centcom ID 
- Centcom Headset
- Departmental Backpack (Sec, Engi or Med)
- Pulse Pistol (10 Pulse, 20 Kill/Stun shots)

There is also a miscellaneous equipment room, this room holds the following:
- Ion Gun
- Box of EMP Grenades
- Box of spare handcuffs
- 4x High Cap Cells
- 6x Gas Masks
- 2x Fire Extinguishers
- 4x Flashes
- Intelicard

The combined two equipment rooms contain the following mechs:
- Gygax Security Mech (1)
- Ripley Construction and Hauling Mech (1)
- Odysseus Medical Mech (1)
- Marauder Combat Mech (2)

New Weapons

Pulse Carbine
![](https://cloud.githubusercontent.com/assets/6595389/5591803/0e6bbe00-91c8-11e4-9291-f6ab61375442.png)
Exclusive to the Security Response Officer. Can fire either 20 Pulse, 20 Laser or 20 stun rounds before exhausting. Can fit in backpacks.

Pulse Pistol
![](https://cloud.githubusercontent.com/assets/6595389/5591805/20b2513c-91c8-11e4-9957-d9130d7ccea9.png)
All ERT Officers have one of these by default barring the Commander. Can fire either 10 Pulse, 20 Laser or 20 Stun rounds before exhausting. Can fit in backpacks and pockets.

---

Please leave suggestions and feedback, thank you.

Steelpoint committed:
EmergencyResponseTeam

Steelpoint committed:
SwapToBaysERTSprites+CommanderSuit

Steelpoint commented:
Difference is the Pulse Rifles predominately use the Destroy setting, as such it makes sense to allow them to use the darker blue setting.

While I would go through and make different colour sprites for the different fire modes, that would take up A LOT of space for each of the three gun variants, also since you indicated you'll make a overlay system I'll simply leave that to you.

---

I'll also add a ERT Commander's room with the Commander armour as well, and work on that repathing.

tkdrg commented:
This will stay on hold until carn is done with her map edits. I'll tag it Do not merge as well. Please remind me to remove the tag once she's done.

Steelpoint committed:
PulseRifleVarients+ArmouryRooms+FixesAndRefinements

Steelpoint committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into EmergencyResponseTeam

Conflicts:
	icons/obj/gun.dmi

Steelpoint commented:
That's cool, as I stated earlier I put this up mainly go garner feedback.

Here's how the new Death Squad area looks like, my expectation is that this would replace the "Click to get Death Squad" button we have and simply make the ERT the go to unit if you want the Death Squad. 

![screenshot 2015-01-01 13 39 40](https://cloud.githubusercontent.com/assets/6595389/5591728/df05c64e-91bb-11e4-9951-ba0dd5138e60.png)

However that's beyond my coding capabilities. The only way into either the "ERT Armoury" or the "Death Squad Equipment Room" is if a admin removes the blast door. I also removed the Secuirty Officers C4 charge since they could use that to blow open the blast doors.

Steelpoint committed:
ReaddPulseGunVarients

Steelpoint commented:
Oh, I'm also having difficulty in giving the ERT Officers ID cards the "NT" symbol. If anyone knows how to do that I would be grateful.

Steelpoint commented:
Minor additions.

* Slightly changed the Pulse Carbine and Pulse Pistol sprites, made them look more refined.
 ![spulc](https://cloud.githubusercontent.com/assets/6595389/5591803/0e6bbe00-91c8-11e4-9291-f6ab61375442.png)  ![spulp](https://cloud.githubusercontent.com/assets/6595389/5591805/20b2513c-91c8-11e4-9957-d9130d7ccea9.png)

 * Pulse Carbine is now a weight 3 item. Can fit in backpacks
 * Pulse Pistol is now a weight 2 item. Can fit in pockets.

* All ERT Officer's closets now come with a Centcom headset, giving them access to not only all station channels, but the deathsquads channel for better communication.

Steelpoint committed:
MinorSpriteFix+CentcomHeadset+GunWeightChange

Nienhaus commented:
That pistol is sexy. I want it.

Steelpoint commented:
Paprika seemed interested in giving a Pulse Pistol to the Head of Security. Which would not be a bad idea as its far easier to balance out a Pistol variant.

Steelpoint commented:
I'm not so sure on what equipment the Commander should get. Currently this is what he has:

* Intelicard
* Mataba(Autorevolver) + 2 speedloaders
* Stun Baton
* Box of handcuffs
* First Aid Kit
* Flash
* One Adminordrazine Medipen

In addition, I've also revisited the equipment rooms, replaced them with the two tile glass airlocks and removed access restrictions to the ERT equipment rooms. I've also given some extra optional equipment to the Sec and Engi officer (Magboots, more metal, night vision).

Steelpoint committed:
ERTCommander+RipleyMech+AdditionEquipment

Alek2ander commented:
I think their heads are oversized. Would removing the "ears" from the helmets would look better?

TZK13 commented:
That's actually what the "eng-ert update" sprite I posted up near the top is supposed to be, the Bay ERT helmet with the "ears" removed.

fj45 commented:
Aren't  those just helmet lights?

Steelpoint commented:
They are headlights, currently I have not implemented that feature. Maybe I should.

fj45 commented:
I would recommend it, otherwise everyone is either forced to grab NVGs or inventory manage nicked flashlights.

Steelpoint committed:
ERTHelmetLights+MinorERTAreaFix

Steelpoint commented:
All good, all the helmets have a inbuilt light.

Steelpoint commented:
I'm going to replace the admin medipen with a combat injector full of the chemical. I might also hand out a adrenal implant for each officer, yes/no?

Steelpoint commented:
Cool, the ERT Med Officer now has a Combat Injector of 100 units of adminordazine. I should note that a mere 10 units of this chemical will heal around 125 health of damage over time.

paprka commented:
Awesome, I hope some admins use these guys, they look sweet as fuck. Kind of like more colorful ODSTs.

Steelpoint committed:
AdminordrazineCombatInjector

Niknakflak commented:
ERTs are the team that goes in and tries to help people and clean up the impossible situation.
Death squads are the team that goes in and purges everything.
Slight difference.

paprka commented:
Honestly a lot of the time death squads usually do the ERT thing anyway and don't kill everyone, which is a bummer.

Steelpoint commented:
Its not uncommon for admins to use the Deathsquad as a ERT simply because no other option exists.

Steelpoint committed:
NTSymbolForERTOfficers

Steelpoint commented:
ERT Officers ID cards will now display a "NT" symbol to anyone wearing a sechud.

TZK13 commented:
There's still red cross imagery on the north, east and west facing views of the medical ERT suit. Also the white square with the red just removed looks a tad out of place, why not just replace the red with blue?

Steelpoint committed:
WhoopsMissedAFewRedCrosses

Steelpoint commented:
Missed those, thanks.

TZK13 commented:
Now the item sprite is missing the little blue bits :^)

TheVekter commented:
:+1: ODST button, would slap many times

paprka commented:
ERT spawns too many people as well, I think 4 total should be an option by the way, one from each 'division'.

Steelpoint commented:
There's no button, its entirely up to a admin to manually bring in players.

There's equipment for seven but that does not mean there's a requirement of there being seven people, as you said though there's the expectation that there is a minimum amount of four officers.

paprka commented:
Oh, uh, you should make a button for this, there's no way admins are going to ever use this if it has a long setup time, that's why we removed death squads before

Steelpoint commented:
Repeating what I said earlier. "Implying I can code. Its all a manual spawn, admins have to get ghosts and respawn them in game currently"

paprka commented:
https://github.com/tgstation/-tg-station/blob/master/code/modules/admin/verbs/debug.dm#L623

This would be fine if you could select their gear out of this list

Steelpoint commented:
I know how to do that, but the way the ERT room is set up is that you just need to spawn in the naked players and they can get their equipment from the lockers. I even did toy with the "select equipment" idea but I decided its far quicker to just spawn in naked and get your stuff.

paprka commented:
It's not though, because you can dress up humans quickly before you even put ghosts in them as an admin. Waiting for newer players tp get all their gear on and stuff is a waste of time for an 'emergency response team' which should be ready once the admins select players.

Trust me, this is, again, the reason death squads were never used. Too long to set up.

Fuck you could even place down mobs that are already pre-equipped with gear and admins wouldn't even have to select their gear.

Preplaced mobs that already have gear or gear selection > cute armory that will never fucking be used

hornygranny commented:
don't merge this if there's no automation

Steelpoint commented:
Having an armoury gives choice to the operatives and allows them to more easily see their equipment, select their loadout, and swap items with other operatives.

Also, there's a difference between admins being hesitant in spawning a Deathsquad, a team that kills everything, and a ERT, a team that helps people.

Jalleo commented:
How about having it mostly pre spawned but specific items you have a choice for by using a voucher on a specific vendor. Akin to the mining vendor with mining vouchers.

TheVekter commented:
For once, I agree with HG. This will be too big of a pain in the ass to set
up. I'm fine with them having to kit up like nuke ops, but at least have a
button to spawn them.
On Jan 5, 2015 4:31 AM, "Jalleo" <notifications@github.com> wrote:

> How about having it mostly pre spawned but specific items you have a
> choice for by using a voucher on a specific vendor. Akin to the mining
> vendor with mining vouchers.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6705#issuecomment-68690984>
> .
>

Jalleo commented:
There is a button which can be changed a tiny bit to have them spawn with all the normal stuff and if reworked properly it could be done quite well.

TheVekter commented:
I would want them to spawn and have a choice of kit, then be given the
chance to customize it a bit. Same as deathsquad.
On Jan 5, 2015 4:42 AM, "Jalleo" <notifications@github.com> wrote:

> There is a button which can be changed a tiny bit to have them spawn with
> all the normal stuff and if reworked properly it could be done quite well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6705#issuecomment-68692165>
> .
>

Steelpoint commented:
Give me a bit.

Steelpoint commented:
Ok, so I have a basic "antag_spawn" system set up. Its based on the Death Squad spawn code but it works.

I'll work on it and elaborate on it a bit more soon.

Steelpoint commented:
Cool.

Current system is that up to seven ghosted players will be selected to spawn in as a ert unit, I've got it set so you can have as few as four players but only a maximum of seven.

Jalleo commented:
Can it be setup to have a selection process of if someone wants to specifically be a certain ERT member or the person dosent mind.

Steelpoint commented:
I'm not skilled or knowledgeable enough to do that.

The only problem I can see is that there's no huge incentive to play as a ERT Commander. Though I could possible force one of the spawning in players to be the Commander.

Jalleo commented:
That is what I was thinking akin to forced captain.

Steelpoint commented:
One of the players will be forced into the commanders role, they will get a com ID card and a backpack, everyone else gets nothing but can get a ID/Pack from the lockers.

Steelpoint committed:
EmergencyResponseTeam-AntagSelectionForAdmins

Steelpoint commented:
I'll do a extensive changelog as well, and the wiki page after this gets merged (if it does).

Steelpoint committed:
MissedANumber

Steelpoint committed:
NameTweaking+VisibleLoyaltyImplant+MovingSecOfficersSecHUDToObviousLocation

paprka commented:
Don't make systems you aren't robust enough to code just so other people can make it less shit

Steelpoint committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into EmergencyResponseTeam

Conflicts:
	_maps/map_files/generic/z2.dmm
	code/_globalvars/lists/mapping.dm

Steelpoint committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into EmergencyResponseTeam

Steelpoint committed:
Updating

Steelpoint commented:
Soooo, is this good to merge or what? I don't think the failed check is my fault as any map related change seems to be affected.

Steelpoint committed:
Changelog

Steelpoint committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into EmergencyResponseTeam

tkdrg commented on html/changelog.html:
```diff
@@ -55,6 +55,16 @@
+				<li class='rscadd'>Two new Pulse gun variants have been added to the game. They are the Pulse Carbine and the Pulse Pistol, both have significantly smaller ammunition capacities than the Pulse Rifle, however they both can fit in a backpack and the pistol can fit in a pocket. All ERT Officers have a Pulse Pistol for a sidearm, however the Security Officers get a Pulse Carbine as their primary  ...
+			</ul>
+		</div>
+
```
Don't do this directly. Make a .yml file instead.

JJRcop replied:
https://github.com/tgstation/-tg-station/blob/master/html/changelogs/example.yml

Steelpoint replied:
I followed the given instructions but whatever.

How do I revert the changelog then.

tkdrg commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -56,7 +56,14 @@
+/obj/item/weapon/reagent_containers/hypospray/combat/adminordrazine/New()
+	..()
+	reagents.remove_reagent("synaptizine", 30)
+	reagents.add_reagent("adminordrazine", 100)
```
I'm not sure how I feel about this. 100 units of adminordrazine is pretty ridiculous, but then again so are most other ERT items, so...

Steelpoint replied:
That's 10 use's on 10 people. If we are sending in the ERT we can assume the station is utterly screwed and medbay may well be gone.

Miauw62 replied:
Adminordrazane is ridiculously powerful, ten units is probably enough to
heal anybody to full health and give them a lot of stun immunity. Imo it
should be a lot of tricord or DD instead.
Op 7-jan.-2015 05:16 schreef "Steelpoint" <notifications@github.com>:

> In code/modules/reagents/reagent_containers/hypospray.dm
> <https://github.com/tgstation/-tg-station/pull/6705#discussion-diff-22570413>
> :
>
> >
> > +/obj/item/weapon/reagent_containers/hypospray/combat/adminordrazine/New()
> > +	..()
> > +	reagents.remove_reagent("synaptizine", 30)
> > +	reagents.add_reagent("adminordrazine", 100)
>
> That's 10 use's on 10 people. If we are sending in the ERT we can assume
> the station is utterly screwed and medbay may well be gone.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6705/files#r22570413>.
>

Steelpoint replied:
It heals about 130 health and stops oxyloss.

hornygranny commented:
i don't see anything that tells me this isn't a second deathsquad

tkdrg commented:
To be honest, I don't really like this, as well. You're making a huge addition to the map and copy-pasting a lot of death squad code to create what is basically reskinned OP non-death squad. I'd rather see these implemented as an alt uniform for death squaddies.

TZK13 commented:
Deathsquads only get gear for murdering, the ERT includes personnel specialized for healing and building.

The difference in intent between what a DS and ERT is supposed to be is what matters here, the entire concept of the DS has been corrupted by admins that make them the equivalent of an ERT which is basically making them pussies.

Steelpoint committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into EmergencyResponseTeam

Steelpoint commented:
There is no merge conflict on my end, this is getting annoying.

The Deathsquad is fully equipped to kill the entire station several times over, they are the most powerful force in the game and they are only meant to kill everything. As noted above their purpose has been corrupted due to admins sending in the Deathsquad to not kill people but help the station, which is not their goal.

The ERT is meant to help, their equipment shows that. Their guns are robust but not meant for long engagement and they take 7 minutes to recharge.

Steelpoint commented:
If you think some gear is overpowered please elaborate and we can look at it.

In my opinion
* Pulse Weapons: Balanced due to their very small ammo capacity and that they take seven minutes to recharge.
* Military RCD: Just a RCD with double the matter number.
* Combat Injector: Powerful, but necessary. Asking a Medical Officer to heal dozens of people with a first aid kit would result in them running out of supplies very fast. A Combat Injector is needed to administer healing to many people.

Steelpoint committed:
Change

Steelpoint commented:
Merge conflicts fixed.

Steelpoint commented:
Tricord and DD are ineffective, the admin drug works well as a admin spawn able only item for the ERT. It is limited in number and only lasts for so long.

I would love to get this merged or rejected so, will this accepted as it stands or what changes do any contributor want me to effect. If you want I can reduce the Combat Injectors admin drug count if you think the 10 uses is too much.

Jalleo commented:
How about changing it so the deathsquad is a more OP alt of the ERT?

Steelpoint commented:
The Deathsquad is still more powerful than the Emergency Response Squad when it comes to wiping out the entire station.

Jalleo commented:
I was meaning code wise merging it all down where possible and making it more easy for the admin to choose stuff.

Steelpoint commented:
Deathsquad and Emergency Response Team spawns are still seperate.

The only difference I can make is to get rid of the normal DS spawn and spawn them in the DS equipment room. 

But I would do that in another PR.

Steelpoint committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into EmergencyResponseTeam

Conflicts:
	icons/mob/head.dmi
	icons/obj/clothing/hats.dmi

Steelpoint committed:
MergeConflict

Steelpoint committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into EmergencyResponseTeam

Conflicts:
	icons/mob/head.dmi
	icons/obj/clothing/hats.dmi

Steelpoint committed:
MergeConflict

Steelpoint commented:
Merge conflicts can bite me.

Steelpoint commented:
Well?

Steelpoint committed:
Nanites!

Steelpoint commented:
@hornygranny, @Miauw62, @MrPerson, @tkdrg, @phil235 or anyone else I forgot. I

This is ready to merge and I talked on IRC about it. Can we merge this please?

Jalleo commented:
does the admin drug heal everything instantly? If so having 10 uses for that is insane I would rather it be a bit of a buff healing thing for 10 uses especially if each person spawns with some.

Steelpoint commented:
There seems to be a LOT of misinformation regarding the drug. Let me state one and for all exactly what 10 units does.

It heals, over time, 130 health. It also cancels all stuns. That's it, nothing else happens.

Jalleo commented:
for how many units of the drug does the 130 health heal?
Also cancelling all stuns are really OP I know this is for ERT but that sounds more like a drug for Deathsquad not ERT give Deathsquad this version and give the ERT a weaker version that dosent cancel stuns. I know you want ERT to be cool but this would make them MORE OP than deathsquad who get no healing stuff.

Steelpoint commented:
As I said, 10 units heal 130hp.

The Deathsquad are very overpowered, their guns, armour, shields, bombs, mechs and numbers make them the most powerful attack force in the game.

The Emergency Response Team lacks that powerful killing equipment, but makes up for it with a good selection of healing and construction items. The polar opposite of the Deathsquad.

Steelpoint committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into EmergencyResponseTeam

Conflicts:
	code/modules/projectiles/guns/energy/nuclear.dm
	icons/obj/gun.dmi

Steelpoint committed:
MergeConflictsV3

Steelpoint commented:
What do I have to do to get a answer on this being merged or not. Cause these merge conflicts keep piling up.

paprka commented:
lel

Steelpoint committed:
QuickTidyUp

Steelpoint commented:
Any progress? I can't work on anything else until this is resolved.

paprka commented:
Yeah someone should merge this it's fine as a concept and 10 days is too long for something this big that edits commonly-used .dmis to sit rotting

@tkdrg pls save this

Miauw62 commented:
Just FYI I don't like a bunch of things about this, but I guess somebody
else will have to tweak this. Don't get hurt if your codebabby gets
touched.
Op 10-jan.-2015 05:55 schreef "paprka" <notifications@github.com>:

> Yeah someone should merge this it's fine as a concept and 10 days is too
> long for something this big that edits commonly-used .dmis to sit rotting
>
> @tkdrg <https://github.com/tkdrg> pls save this
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6705#issuecomment-69440905>
> .
>

paprka commented:
Brb making revert PR

Nah but it can be changed later

/sweeps feature under the rug

Steelpoint commented:
If there's anything in particular that you think needs adjusting please mention it either here or on the feedback thread on the forums. 

I'll also see how it goes in game.

Carn commented:
This reverted a number of map fixes for shuttles
:sad:
